src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/ElfObjectWriter.cs

```diff
@@ -761,8 +801,12 @@ private void EmitObjectFile<TSize>(FileStream outputFileStream)
                 SectionHeaderEntrySize = (ushort)ElfSectionHeader.GetSize<TSize>(),
                 SectionHeaderEntryCount = sectionCount < SHN_LORESERVE ? (ushort)sectionCount : (ushort)0u,
                 StringTableIndex = strTabSectionIndex < SHN_LORESERVE ? (ushort)strTabSectionIndex : (ushort)SHN_XINDEX,
-                // For ARM32 claim conformance with the EABI specification
-                Flags = _machine is EM_ARM ? 0x05000000u : 0u,
+                Flags = _machine switch
+                {
+                    EM_ARM => 0x05000000u, // For ARM32 claim conformance with the EABI specification
+                    EM_LOONGARCH => 0x43u, // For LoongArch ELF psABI specify the ABI version (1) and modifiers (64-bit GPRs, 64-bit FPRs)
+                    _ => 0u
+                },
```

Comment:
nit - maybe a bit readable this way:

```c#
Flags = _machine switch
{
    EM_ARM => (0x05u /* ABI version */ << 24),
    EM_LOONGARCH => (0x1u /* ABI version */ << 6) | 0x3u /* modifiers */,
    _ => 0u
};
```

Reply:
I prefer to keep the reference to the spec so it’s searchable. I am fine with adding bit shifts but don’t consider it necessary.

Reply:
Updated with comments. The comment was referring to version and modifier so I was thinking maybe it's better to lay out how it was formed. (no diff in codegen etc.)